### ISSUE TITLE
German umlaute not shown

### ISSUE THREAD

**Issue by ThomasKrenn** [JakubMelka/PDF4QT#59]
Hello Jakub,
please find enclosed a document created with MS Word, that contains german umlaute that are not displayed.

A second issue is that the page size on screen is not correct.

The size of the document is: 8,27in x 11,69in

Displayed with 100% on a 96dpi monitor (Win default) the size on screen should be:
794px  x  1122px

Best regards
Thomas

[zottegem.pdf](https://github.com/JakubMelka/PDF4QT/files/11934386/zottegem.pdf)

**Comment by JakubMelka** [comment]
For @ThomasKrenn: I have fixed the character font issue.

However, I cannot simulate the size of the document issue. The monitor must support it (provide correct information about the **physical** monitor size). Not each monitor can do it.

I have tried it on my 4K monitor, and then place a physical sheet of A4 paper onto it and it matched perfectly.

**Comment by ThomasKrenn** [comment]
Hello @JakubMelka 
thanks for your feedback.

On screen the "logical" correctness is important. Qt provides the: logicalDotsPerInch

[logicalDotsPerInch](https://doc.qt.io/qt-6/qscreen.html#logicalDotsPerInch-prop)

Here you can find an old sample program: 
[qscreen-class](https://www.ics.com/blog/whats-new-qt-5-qscreen-class)

To check if the size is "logically" correct you can use a screen ruler. 

E.g.  [Win Screenruler](https://sourceforge.net/projects/screenruler/)

BTW: When rendering the document to PNG you must set: **png_set_pHYs**

Best regards
Thomas

**Comment by JakubMelka** [comment]
To @ThomasKrenn:
For certain applications, such as CAD drawings, it is essential to utilize physical pixels to ensure that the scale on the screen matches that on paper. However, I also recognize the validity of your argument regarding the inclusion of logical dpi. Consequently, I have implemented a new feature in the options menu that allows users to enable logical dpi for accurate document zoom calculations.
![image](https://github.com/JakubMelka/PDF4QT/assets/11367048/5a8037c3-6b2e-4953-a136-5edb09c2426d)

**Comment by ThomasKrenn** [comment]
Hello @JakubMelka ,
thanks for your feedback.

Maybe you should consider making the logical resolution the default. I am positive that this is what most users expect as default. (Displaying the document using the phys. screen resolution is a cool feature. No other viewer I know has it.)


#### Rendering to PNG (Render to images' in the file menu'):

In 'PDFRenderToImagesDialog' please set the resolution of the rendered image using **setDotsPerMeterX** and **setDotsPerMeterY**.

Qt will then call **png_set_pHYs** when creating the PNG:

```
     if (image.dotsPerMeterX() > 0 || image.dotsPerMeterY() > 0) {
        png_set_pHYs(png_ptr, info_ptr,
                image.dotsPerMeterX(), image.dotsPerMeterY(),
                PNG_RESOLUTION_METER);
```    

See also:
[PNG Options](http://www.libpng.org/pub/png/book/chapter11.html#png.ch11.div.8)

**Comment by JakubMelka** [comment]
@ThomasKrenn : I have fixed it. Now DPM is being set. I did it in the rasterizer, which creates images.

**Comment by JakubMelka** [comment]
The issue has been resolved in version 1.3.4. Should you believe the problem persists, please do not hesitate to reopen the issue.
